pkg/framework/machines.go

```diff
@@ -20,17 +20,23 @@ import (
 )
 
 // FilterRunningMachines returns a slice of only those Machines in the input
-// that are in the "Running" phase.
-func FilterRunningMachines(machines []*mapiv1beta1.Machine) []*mapiv1beta1.Machine {
+// that are in the "Running" phase. If some of the machine enters Failed phase,
+// it will indicate this wil an error.
+func FilterRunningMachines(machines []*mapiv1beta1.Machine) ([]*mapiv1beta1.Machine, error) {
 	var result []*mapiv1beta1.Machine
 
 	for i, m := range machines {
-		if m.Status.Phase != nil && *m.Status.Phase == MachinePhaseRunning {
-			result = append(result, machines[i])
+		if m.Status.Phase != nil {
+			switch *m.Status.Phase {
+			case MachinePhaseRunning:
+				result = append(result, machines[i])
+			case MachinePhaseFailed:
+				return nil, fmt.Errorf("Machine entered the Failed phase: %q, reason: %v", m.GetName(), m.Status.ErrorMessage)
+			}
```

Comment:
One question that comes to mind is whether any of the tests rely on a Machine going into a failed phase, we need to make sure we aren't breaking those by doing this

Reply:
We are not testing that at the moment, so it is not a breaking change. Why wait for machines running if you are expecting them not to?

Reply:
Good question 😅  I just wondered if we were using this to check that a machine didn't become running. I think it's ok